### PR TITLE
Allow to use our logger code in external libraries

### PR DESCRIPTION
**Pull request by TheOneRing**
I removed the install target for probably not working headers and added the logger.h file instead.
Ideally we would also provide a cmake export target but at the moment I would not recommend anybody to use the logger besides for testing purposes.

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11978/6/1

<details>


</details>

File: src/libsync/CMakeLists.txt [hunk 1]
```diff
@@ -69,31 +69,15 @@ endif()
...
+    ${CMAKE_CURRENT_BINARY_DIR}/owncloudlib.h
+    logger.h
 )
 
-set(creds_HEADERS
-    creds/abstractcredentials.h
-    creds/httpcredentials.h
+INSTALL(
+    FILES ${owncloudsync_HEADERS}
+    DESTINATION ${INCLUDE_INSTALL_DIR}/${synclib_NAME}
```
Comment (fmoc) [inline src/libsync/CMakeLists.txt:78]: You should probably add these to the target, then use `install(TARGETS owncloudsync [PUBLIC|PRIVATE]_HEADER DESTINATION <...>)`.

**Review by fmoc — APPROVED** [review]
Added a suggestion, but I think since this PR just moves some code around, there's no need to perform this refactoring right now. We use plenty legacy APIs in CMake anyway.

File: src/libsync/CMakeLists.txt (hunk 1 above)
Comment (TheOneRing) [reply src/libsync/CMakeLists.txt:78]: As far as I understand PUBLIC_HEADER is a mac os framework thingy.

**Pull request merged**